omnipaxos/src/sequence_paxos/follower.rs

```diff
@@ -174,41 +174,34 @@ where
                     }
                 }
             };
-            self.state = (Role::Follower, Phase::Accept);
-            self.current_seq_num = accsync.seq_num;
-            let cached_idx = self.outgoing.len();
-            self.latest_accepted_meta = Some((accsync.n, cached_idx));
-            self.outgoing.push(PaxosMessage {
-                from: self.pid,
-                to: from,
-                msg: PaxosMsg::Accepted(accepted),
-            });
             match accsync.stopsign {
                 Some(ss) => {
-                    if let Some(ss_entry) = self
-                        .internal_storage
-                        .get_stopsign()
-                        .expect("storage error while trying to read stopsign")
-                    {
-                        let StopSignEntry {
-                            decided: has_decided,
-                            stopsign: _my_ss,
-                        } = ss_entry;
-                        if !has_decided {
+                    if let Some(ss_entry) = self.internal_storage.get_stopsign() {
+                        if !ss_entry.decided(old_decided_idx) {
                             self.accept_stopsign(ss);
                         }
                     } else {
                         self.accept_stopsign(ss);
                     }
-                    let a = AcceptedStopSign { n: accsync.n };
-                    self.outgoing.push(PaxosMessage {
-                        from: self.pid,
-                        to: from,
-                        msg: PaxosMsg::AcceptedStopSign(a),
-                    });
+                    let ss_idx = self
+                        .internal_storage
+                        .get_stopsign()
+                        .expect("stopsign should have been accepted")
+                        .log_idx;
+                    assert_eq!(ss_idx, accepted.accepted_idx);
+                    accepted.accepted_idx = ss_idx + 1;
```

Comment:
Unify the access to accepted index by always calling the `get_accepted_idx()`

```suggestion
                    accepted.accepted_idx = self.internal_storage.get_accepted_idx();
```